app/src/main/java/com/mealsmadeeasy/ui/meal/MealActivity.kt

```diff
@@ -42,31 +44,68 @@ class MealActivity : BaseActivity() {
         mealStore.findMealById(intent.getStringExtra("meal_id"))
                 .bindToLifecycle(this)
                 .subscribe({meal ->
-                    finishSetup(meal)
+                    mealSetup(meal)
                 }, { throwable ->
-                    Log.e(TAG, "Failed to load suggestions", throwable)
+                    Log.e(TAG, "Failed to load meal", throwable)
                     Toast.makeText(this, R.string.meal_page_failed_to_load_meal, Toast.LENGTH_SHORT).show()
                 })
     }
 
-    private fun finishSetup(meal: Meal) {
+    private fun mealSetup(meal: Meal) {
         supportActionBar?.title = meal.name
 
         val thumbnailView = findViewById<ImageView>(R.id.meal_thumbnail)
         val nameView = findViewById<TextView>(R.id.meal_name)
-        val descriptionView = findViewById<TextView>(R.id.meal_description)
         Picasso.with(this)
                 .load(meal.thumbnailUrl)
                 .fit()
                 .centerCrop()
                 .into(thumbnailView)
         nameView.text = meal.name
-        descriptionView.text = meal.description
 
         val addButton = findViewById<FloatingActionButton>(R.id.meal_add_to_plan)
         addButton.setOnClickListener {
             startActivity(AddToPlanActivity.newIntent(this, meal.id))
         }
+
+        mealStore.getRecipe(meal.id)
+                .bindToLifecycle(this)
+                .subscribe({recipe ->
+                    recipeSetup(recipe)
+                }, { throwable ->
+                    Log.e(TAG, "Failed to load recipe", throwable)
+                    Toast.makeText(this, R.string.meal_page_failed_to_load_recipe, Toast.LENGTH_SHORT).show()
+                })
+    }
+
+    private fun recipeSetup(recipe: Recipe) {
+        val prepView = findViewById<TextView>(R.id.meal_page_prep_time)
+        val stepsView = findViewById<TextView>(R.id.meal_page_steps)
+        val ingredientsView = findViewById<TextView>(R.id.meal_page_ingredients)
+
+        val prepTime = recipe.prepTime.toString() + " " + getString(R.string.meal_page_minutes)
+        prepView.text = prepTime
+
+        val newLine = System.getProperty("line.separator")
```

Comment:
Whoa. What are you trying to do with this? I feel like we can do something better in the layout xml.